tests/test_user_profile.py

```diff
@@ -91,3 +95,30 @@ def test_read_profile_for_player(self):
             datetime(2021, 2, 12, 0, 15, 0): 0.5,
             datetime(2021, 2, 12, 0, 30, 0): 0.1
         })
+
+    def test_correct_time_expansion_read_arbitrary_profile(self):
+        market_maker_rate = 30
+        if GlobalConfig.IS_CANARY_NETWORK:
+            GlobalConfig.sim_duration = duration(hours=3)
+            expected_last_time_slot = today(tz=TIME_ZONE).add(days=PROFILE_EXPANSION_DAYS - 1,
+                                                              hours=23, minutes=45)
+            mmr = read_arbitrary_profile(InputProfileTypes.IDENTITY, market_maker_rate)
+            assert list(mmr.keys())[-1] == expected_last_time_slot
+            GlobalConfig.sim_duration = duration(hours=30)
+            expected_last_time_slot = today(tz=TIME_ZONE).add(days=PROFILE_EXPANSION_DAYS - 1,
+                                                              hours=23, minutes=45)
+            mmr = read_arbitrary_profile(InputProfileTypes.IDENTITY, market_maker_rate)
+            assert list(mmr.keys())[-1] == expected_last_time_slot
+        else:
+            GlobalConfig.FUTURE_MARKET_DURATION_HOURS = 0
+            GlobalConfig.sim_duration = duration(hours=3)
+            mmr = read_arbitrary_profile(InputProfileTypes.IDENTITY, market_maker_rate)
+            assert (list(mmr.keys())[-1] - today(tz=TIME_ZONE)).days == 0
+            GlobalConfig.sim_duration = duration(hours=36)
+            mmr = read_arbitrary_profile(InputProfileTypes.IDENTITY, market_maker_rate)
+            assert (list(mmr.keys())[-1] - today(tz=TIME_ZONE)).days == 1
```

Comment:
How about being even more and check for the hours in order to catch (unlikely) errors that only add a few future markets.
```suggestion
            assert (list(mmr.keys())[-1] - today(tz=TIME_ZONE)).hours == 24
```
Also applies to the other checks.

Reply:
It does not work like that. Pendulum duration(days=1) has the following members:
duration.hours = 0
duration.days = 1
In order to satisfy it I will save it in a variable and compare both

Reply:
Darn, I always confuse this. I actually meant:
```suggestion
            assert (list(mmr.keys())[-1] - today(tz=TIME_ZONE)).total_hours == 24
```